src/apps/util/config.clj

```diff
@@ -240,55 +241,55 @@
        (cfg/env-setting "DATA_INFO_PORT")
        (data-info-base-url)))))
 
-(cc/defprop-str notification-agent-base
+(cc/defprop-optstr notification-agent-base
   "The base URL for the notification agent."
   [props config-valid configs]
-  "apps.notificationagent.base-url")
+  "apps.notificationagent.base-url" "http://notification-agent:60000")
 
-(cc/defprop-str ipg-base
+(cc/defprop-optstr ipg-base
   "The base URL for the iplant-groups service."
   [props config-valid configs]
-  "apps.iplant-groups.base-url")
+  "apps.iplant-groups.base-url" "http://iplant-groups:60000")
 
-(cc/defprop-str de-grouper-user
+(cc/defprop-optstr de-grouper-user
   "The username that the DE uses to authenticate to Grouper."
   [props config-valid configs]
-  "apps.iplant-groups.grouper-user")
+  "apps.iplant-groups.grouper-user" "de_grouper")
 
-(cc/defprop-str metadata-base
+(cc/defprop-optstr metadata-base
   "The base URL for the metadata service."
   [props config-valid configs]
-  "apps.metadata.base-url")
+  "apps.metadata.base-url" "http://metadata:60000")
 
-(cc/defprop-int job-status-poll-interval
+(cc/defprop-optint job-status-poll-interval
   "The job status polling interval in minutes."
   [props config-valid configs]
-  "apps.jobs.poll-interval")
+  "apps.jobs.poll-interval" 15)
 
-(cc/defprop-str permissions-base
+(cc/defprop-optstr permissions-base
   "The base URL for the permissions service."
   [props config-valid configs]
-  "apps.permissions.base-url")
+  "apps.permissions.base-url" "http://permissions:60000")
 
 (cc/defprop-str ui-base-url
   "The base URL for the email service."
   [props config-valid configs]
   "apps.ui.base-url")
 
-(cc/defprop-str iplant-email-base-url
+(cc/defprop-optstr iplant-email-base-url
   "The base URL for the email service."
   [props config-valid configs]
-  "apps.email.base-url")
+  "apps.email.base-url" "http://iplant-email:60000")
 
-(cc/defprop-str app-deletion-notification-src-addr
+(cc/defprop-optstr app-deletion-notification-src-addr
   "The source email address of app deletion notification messages."
   [props config-valid configs]
-  "apps.email.app-deletion.from")
+  "apps.email.app-deletion.from" "prod-user@cyverse.org")
```

Comment:
Not sure about having this default email coming from our domain. This means everyone who runs their own DE will be sending app deletion messages to their users from this address.
The current `ansible/inventories/group_vars/all` just defaults this address to a blank string, but I'm not sure if that's appropriate either.
Maybe this shouldn't have any default? Or something more obvious, like `de-notifications@example.com` or `no-reply-please-update-your-de-apps-config@cyverse.org`? 😜